### PR TITLE
Remove log group from AWS batch system test

### PR DESCRIPTION
**Pull request by vincbeck**
Whenever a new batch job is executed, a new Cloudwatch log stream is created. We should remove this log stream as part of the test as well.

### CODE REVIEW

File: tests/system/providers/amazon/aws/example_batch.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 from datetime import datetime
+from typing import List, Optional, Tuple
```
Comment (pankajkoti) [inline tests/system/providers/amazon/aws/example_batch.py]: Can you try using `__future__.annotations` like in PR: https://github.com/apache/airflow/pull/26290/files

**Review by pankajkoti — APPROVED** [review]
Looks good to me, minor suggestion for the imports.

File: tests/system/providers/amazon/aws/example_batch.py (hunk 1 above)
Comment (potiuk) [reply tests/system/providers/amazon/aws/example_batch.py]: I think this might be better done after I merge it (today :))

Comment (vincbeck) [reply tests/system/providers/amazon/aws/example_batch.py]: Done

**Review by potiuk — APPROVED** [review]

**Pull request merged**